library/UnicornHat/unicornhat.py

```diff
@@ -274,6 +276,9 @@ def set_pixel(x, y, r, g=None, b=None):
 
     if type(r) is tuple:
         r, g, b = r
+    
+    elif type(r) is str:
+        r, g, b = colors[r]
```

Comment:
You should include a try/except clause here to trap invalid colours and raise an appropriate error message. At the moment the result is a little confusing:

```
>>> unicornhat.set_pixel(0,0,"sfshf")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/pi/Development/unicorn-hat/library/UnicornHat/unicornhat.py", line 281, in set_pixel
    r, g, b = colors[r]
KeyError: 'sfshf'
```

Along the same lines, it might be worth converting the string value to lowercase before looking up a value in the colors array so that entries like "Green" and "GREEN" will work:

```
>>> unicornhat.set_pixel(0,0,"Green")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/pi/Development/unicorn-hat/library/UnicornHat/unicornhat.py", line 281, in set_pixel
    r, g, b = colors[r]
KeyError: 'Green'
```